compiler/plugins/input/StableHLO/Conversion/LegalizeCHLO.cpp

```diff
@@ -1,2226 +0,0 @@
-  Value x = args.front();
-  assert(cast<ShapedType>(x.getType()).getElementType().isF32() &&
-         "expect f32 element type");
-  const float kAlpha[] = {
-      -2.72614225801306e-10f, 2.77068142495902e-08f,  -2.10102402082508e-06f,
-      -5.69250639462346e-05f, -7.34990630326855e-04f, -2.95459980854025e-03f,
-      -1.60960333262415e-02f,
-  };
-  const float kBeta[] = {
-      -1.45660718464996e-05f, -2.13374055278905e-04f, -1.68282697438203e-03f,
-      -7.37332916720468e-03f, -1.42647390514189e-02f,
-  };
-
-  // Clamp argument between -4 and 4.
-  Value lb = getConstantLike(rewriter, loc, -4.0, x);
-  Value ub = getConstantLike(rewriter, loc, 4.0, x);
-  x = rewriter.create<mlir::stablehlo::ClampOp>(loc, x.getType(), lb, x, ub);
-  Value xSq = rewriter.create<mlir::stablehlo::MulOp>(loc, x, x);
-
-  // Materialize polynomial approximation for x in [-4, 4] as
-  //   erf(x) = x * Alpha(x^2) / Beta(x^2).
-  Value alphaPoly = materializePolynomialApproximation(rewriter, loc, xSq,
-                                                       llvm::ArrayRef(kAlpha));
-  Value betaPoly = materializePolynomialApproximation(rewriter, loc, xSq,
-                                                      llvm::ArrayRef(kBeta));
-  Value xMulAlphaPoly =
-      rewriter.create<mlir::stablehlo::MulOp>(loc, x, alphaPoly);
-  Value erf =
-      rewriter.create<mlir::stablehlo::DivOp>(loc, xMulAlphaPoly, betaPoly);
-  Value lbErf = getConstantLike(rewriter, loc, -1.0, x);
-  Value ubErf = getConstantLike(rewriter, loc, 1.0, x);
-  return rewriter.create<mlir::stablehlo::ClampOp>(loc, erf.getType(), lbErf,
-                                                   erf, ubErf);
-}
-
-static Value
-materializeErfcApproximationF32(ConversionPatternRewriter &rewriter,
-                                Location loc, ValueRange args) {
-  Value x = args.front();
-  assert(cast<ShapedType>(x.getType()).getElementType().isF32() &&
-         "expect f32 element type");
-
-  // Rely on erfc approximation for |x| >= 1
-  //   erfc(x) = erfc_approx(x)
-  Value erfcApprox =
-      materializeErfcApproximationF32ForMagnitudeGeOne(rewriter, loc, x);
-
-  // Rely on erf approximation for |x| < 1 and materialize erfc as
-  //   erfc(x) = 1 - erf_approx(x)
-  Value one = getConstantLike(rewriter, loc, 1.0, x);
-  Value erfApprox =
-      materializeErfApproximationF32ForMagnitudeLeOne(rewriter, loc, x);
-  Value erfBasedApprox =
-      rewriter.create<mlir::stablehlo::SubtractOp>(loc, one, erfApprox);
-
-  // Materialize approximation selection based on argument.
-  Value absX = rewriter.create<mlir::stablehlo::AbsOp>(loc, x);
-  Value absXLtOne = rewriter.create<mlir::stablehlo::CompareOp>(
-      loc, absX, one, mlir::stablehlo::ComparisonDirection::LT);
-  return rewriter.create<mlir::stablehlo::SelectOp>(loc, absXLtOne,
-                                                    erfBasedApprox, erfcApprox);
-}
-
-struct ConvertErfOp final : OpConversionPattern<mlir::chlo::ErfOp> {
```

Comment:
I guess this `chlo.erf` is still being used:
https://github.com/iree-org/iree/actions/runs/11170571951/job/31053680322?pr=18681#step:7:2529
```
1210/1569 Test #1307: iree/tests/e2e/stablehlo_models/mnist_train_test/mnist_train_test_cpu ........................................***Failed    0.88 sec
E
======================================================================
ERROR: test_mnist_training (__main__.MnistTrainTest.test_mnist_training)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/iree/iree/tests/e2e/stablehlo_models/mnist_train_test/mnist_train_test.py", line 122, in test_mnist_training
    module = build_module(tmp_dir)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/__w/iree/iree/tests/e2e/stablehlo_models/mnist_train_test/mnist_train_test.py", line 29, in build_module
    compile_file(
  File "/__w/iree/iree/build/compiler/bindings/python/iree/compiler/tools/core.py", line 255, in compile_file
    result = invoke_immediate(cl)
             ^^^^^^^^^^^^^^^^^^^^
  File "/__w/iree/iree/build/compiler/bindings/python/iree/compiler/tools/binaries.py", line 199, in invoke_immediate
    raise CompilerToolError(process)
iree.compiler.tools.binaries.CompilerToolError: Error invoking IREE compiler tool iree-compile
Error code: 1
Diagnostics:
in.mlir:550:12: error: 'chlo.erf' op using VHLO and StableHLO Ops in the same module is not supported. 
in.mlir:549:12: remark: last StableHLO Op was found here: 
```

We could try regenerating that model input (or reimplementing the test using more recent frameworks/versions): https://github.com/iree-org/iree/blob/a9f562818460a8416165a64765d3fe90283265a8/tests/e2e/stablehlo_models/mnist_train_test/mnist_train_test.py#L22-L34